.github/workflows/e2e-compatibility-workflow-call.yaml

```diff
@@ -0,0 +1,50 @@
+on:
+  workflow_call:
+    inputs:
+      docker-tag:
+        description: 'Docker tag being used'
+        required: true
+        type: string
+      test-suite:
+        description: 'Test suite to run'
+        required: true
+        type: string
+
+
+jobs:
+  load-test-matrix:
+    outputs:
+      test-matrix: ${{ steps.set-test-matrix.outputs.test-matrix }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          # use jq -c to put the full json contents on a single line. This is required when using the json body
+          # to create the matrix in the following job.
+          test_matrix="$(cat .github/compatibility-test-matrices/${{ inputs.docker-tag }}/${{ inputs.test-suite }}.json | jq -c)"
+          echo $test_matrix
+          echo "::set-output name=test-matrix::$test_matrix"
+        id: set-test-matrix
+
+  e2e:
+    runs-on: ubuntu-latest
+    needs: load-test-matrix
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.load-test-matrix.outputs.test-matrix) }}
```

Comment:
Aha, so this magic uses `load-test-matrix` to parse the JSON files and set the values in `matrix` which is used below to run the e2e test and populate the env?

Reply:
yes the previous job `load-test-matrix` stores a json string as its output (the contents of a json file depending on args) and the `matrix` field accepts dynamic input (or statically defined in the workflow)

Reply:
This is indeed magic :sweat_smile:

Reply:
the process of passing variables between steps is quite cumbersome unfortunately 😓